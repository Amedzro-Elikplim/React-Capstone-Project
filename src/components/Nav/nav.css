.nav {
    padding: 10px;
    background-color: #7FD2E4;
    color: white;
}

.button-toggler {
    border: none;
    outline: none;
    color: white;
}

.nav-content {
    padding: 10px;
}

.nav-content li {
    margin-top: 10px;
    border-bottom: 1px solid white;
}