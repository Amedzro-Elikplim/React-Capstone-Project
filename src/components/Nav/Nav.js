import './nav.css';
import React from 'react';
import { Link } from 'react-router-dom';
import { FiMenu } from 'react-icons/fi';

function Nav() {
  return (
    <nav className="nav">
      <span
        type="button"
        className="navbar-toggler button-toggler"
        data-bs-toggle="collapse"
        data-bs-target="#navbarNav"
        aria-controls="navbarNav"
      >
        <FiMenu />
      </span>
      <div className="collapse navbar-collapse nav-content" id="navbarNav">
        <ul className="navbar-nav links">
          <Link to="/" className="nav-item item">Weather</Link>
          <Link to="/pollution" className="nav-item item">Pollution per country</Link>
          <Link to="/map" className="nav-item item">Weather maps</Link>
        </ul>
      </div>
    </nav>
  );
}

export default Nav;
