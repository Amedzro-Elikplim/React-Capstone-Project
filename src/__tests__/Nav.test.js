import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../redux/ConfigureStore';
import Nav from '../components/Nav/Nav';

test('it should render Nav bar', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <HashRouter>
          <Nav />
        </HashRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
