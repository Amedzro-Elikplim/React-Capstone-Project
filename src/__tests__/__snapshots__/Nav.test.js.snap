// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render Nav bar 1`] = `
<nav
  className="nav"
>
  <span
    aria-controls="navbarNav"
    className="navbar-toggler button-toggler"
    data-bs-target="#navbarNav"
    data-bs-toggle="collapse"
    type="button"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="3"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="18"
        y2="18"
      />
    </svg>
  </span>
  <div
    className="collapse navbar-collapse nav-content"
    id="navbarNav"
  >
    <ul
      className="navbar-nav links"
    >
      <a
        className="nav-item item"
        href="#/"
        onClick={[Function]}
      >
        Weather
      </a>
    </ul>
  </div>
</nav>
`;
